doc/src/manual/metaprogramming.md

```diff
@@ -708,6 +729,27 @@ julia> foo()
 
 This kind of manipulation of variables should be used judiciously, but is occasionally quite handy.
 
+Getting the hygiene rules correct can be a formidable challenge,
+Before using a macro, you might want to consider whether a function closure
+would be sufficient. Another useful strategy is to defer as much work as possible to runtime.
+For example, many macros simply wrap their arguments in a QuoteNode or other similar Expr.
+Some examples of this include `@task body` which simply returns `schedule(Task(() -> $body))`,
+and `@eval expr`, which simply returns `eval(QuoteNode(expr))`.
+
+To demonstrate, we might rewrite the `@time` example above as:
+```
+macro time(expr)
+    return :(timeit(() -> $(esc(expr))))
+end
+function timeit(f)
+    t0 = time()
+    val = f()
+    t1 = time()
+    println("elapsed time: ", t1-t0, " seconds")
+    return val
+end
+```
+
```

Comment:
Because that was the point of having functions be first-class types - it's no longer possible to measure the performance impact of calling (exactly) one.